cli/stream_command.go

```diff
@@ -1409,6 +1419,10 @@ func (c *streamCmd) interactiveEdit(cfg api.StreamConfig) (api.StreamConfig, err
 }
 
 func (c *streamCmd) editAction(pc *fisk.ParseContext) error {
+	if (c.repubSource != "" && c.repubDest == "") || (c.repubSource == "" && c.repubDest != "") {
+		fisk.Fatalf("must specify both --republish-source and --republish-destination")
```

Comment:
The problem is this:
https://github.com/nats-io/natscli/blob/84138d64116854a7c954fbb0d9e451657d2d7107/cli/stream_command.go#L1354

I discovered during my testing that if you specify just one of the two, it ignores it (and the user gets a 'configs are the same') (same thing with headers-only BTW). So I just added this guard up front while I was there :).

Ultimately, it's similar as above with subject transform source and dest how to distinguish the missing one of the two as meaning set the existing one to "" or as don't modify the existing one. We could go with the meaning a re-use the current value if not specified, but in the end, to me, I still have the same problem as above with not being able to use edit with command line arguments to remove things - it feels that for anything that you may want to add or remove (rather than just change like a replication degree) using command line arguments feels broken.

Like, how to you remove a tag besides editing the JSON?

Reply:
Same as above I guess with the `IsSetByUser()` then.